brave_domains/service_domains.cc

```diff
@@ -75,7 +75,7 @@ std::string ConvertEnvironmentToString(brave_domains::ServicesEnvironment env) {
     return it->second;
   }
 
-  NOTREACHED_IN_MIGRATION();
+  LOG(ERROR) << "Invalid ServicesEnvironment value: " << static_cast<int>(env);
```

Comment:
I think this should be `NOTREACHED` actually. We own the enum, which is

```cxx
enum ServicesEnvironment { DEV, STAGING, PROD };
```

I particularly think this function should be something like:
```cxx
std::string ConvertEnvironmentToString(brave_domains::ServicesEnvironment env) {
 switch(env) {
    case ServicesEnvironment::DEV:
       return kBraveServicesSwitchValueDev;
    case ServicesEnvironment::STAGING:
       return kBraveServicesSwitchValueStaging;
    case ServicesEnvironment::PROD:
       return kBraveServicesSwitchValueProduction;  
 }
  NOTREACHED_NORETURN();
}
```

This way we don't get some hidden bug if a new key is added to `brave_domains::ServicesEnvironment`.

Reply:
You don't need to do it if you don't feel like, but ideally we should also change this enum from

```cxx
enum ServicesEnvironment { DEV, STAGING, PROD };
```

to:
```cxx
enum class ServicesEnvironment { kDev, kStaging, kProd };
```

So, if you feel like doing that too....